cassandra/components.py

```diff
@@ -650,67 +665,62 @@ def run_component(self):
         import xanthos
 
         config_file = self.params["config_file"]
-
         xth = xanthos.Xanthos(config_file)
 
         args = {}
+        gridded_runoff = []
+
+        # Other components should produce gridded climate data as a list of 2d numpy arrays
+        cap_names = ['gridded_pr', 'gridded_tas', 'gridded_pr_coord', 'gridded_tas_coord']
+        if all(cap in self.cap_tbl for cap in cap_names):
+            pr_grids = self.fetch('gridded_pr')
+            tas_grids = self.fetch('gridded_tas')
+            pr_coord = self.fetch('gridded_pr_coord')
+            tas_coord = self.fetch('gridded_tas_coord')
+        else:
+            pr_grids = tas_grids = []
+            xth_results = xth.execute(args)
+            gridded_runoff.append(xth_results.Q)
 
-        # Wait for other components to produce precipitation and
-        # temperature data (as pandas DataFrames), if provided
-        self.add_args_from_capability(args, 'PrecipitationFile', 'gridded_pr', 'mm_month-1')
-        self.add_args_from_capability(args, 'trn_tas', 'gridded_tas', 'C')
-
-        xth_results = xth.execute(args)
+        # Run Xanthos for each pair of precipitation and temperature grids
+        for pr, tas in zip(pr_grids, tas_grids):
+            args['PrecipitationFile'] = self.prep_for_xanthos(pr, pr_coord)
+            args['trn_tas'] = self.prep_for_xanthos(tas, tas_coord) - 273.15  # K to C
+            xth_results = xth.execute(args)
+            gridded_runoff.append(xth_results.Q)
 
-        # Add runoff (Q) results from xanthos to the gridded_runoff capability
-        self.addresults("gridded_runoff", xth_results.Q)
+        self.addresults("gridded_runoff", gridded_runoff)
```

Comment:
Do we document anywhere the format of this capability?  I guess it's probably a matrix of grid cells and months, but we'd probably also like to know where we can find things like grid cell coordinates, time coordinate, etc.  Basically, when I go to write the drought analysis component next week, where do I find the information I need to interpret the numbers in this array?